### DISCUSSION TITLE
How to compare different devices on the same node?

### DISCUSSION THREAD

**Discussion by zhucan** [louwrentius/fio-plot#117] [Q&A]
bench-fio cmd is:
`./bench-fio --target /dev/vdb /dev/vdc --type device --iodepth 1 8 16 32 64 --numjobs 1 8 --mode randwrite --output TEST --destructive --b 4k  32k 4M`

output like this:
```
TEST/
└── vdb
    ├── 32k
    │   ├── randwrite-1-1.json
    │   ├── randwrite-16-1.json
    │   ├── randwrite-16-8.json
    │   ├── randwrite-1-8.json
    │   ├── randwrite-32-1.json
    │   ├── randwrite-32-8.json
    │   ├── randwrite-64-1.json
    │   ├── randwrite-8-1.json
    │   ├── randwrite-8-8.json
    │   ├── randwrite-iodepth-16-numjobs-1_bw.1.log
    │   ├── randwrite-iodepth-16-numjobs-1_clat.1.log
    │   ├── randwrite-iodepth-16-numjobs-1_iops.1.log
    │   ├── randwrite-iodepth-16-numjobs-1_lat.1.log
    │   ├── randwrite-iodepth-16-numjobs-1_slat.1.log
    │   ├── randwrite-iodepth-16-numjobs-8_bw.1.log
    │   ├── randwrite-iodepth-16-numjobs-8_bw.2.log
    │   ├── randwrite-iodepth-16-numjobs-8_bw.3.log
```

How to compare "/dev/vdb" and "/dev/vdc"?

**Comment by zhucan** [comment]
Compare cmd is:
`./fio-plot -i TEST_VDB/vdb TEST_VDB/vdc  --source "https://louwrentius.com"  -T "Comparing IOPs performance of multiple SSDs" -g -t iops -r randwrite --xlabel-parent 0 -d 1 8 16 32 64 -n 1 8 `

but the png is:
![Comparing-IOPs-performance-of-multiple-SSDs_2023-06-09_181436_VX](https://github.com/louwrentius/fio-plot/assets/13997135/9cf0dd75-9754-42ef-8368-9cb0e29dbf5a)

**Comment by louwrentius** [comment]
You have to play with the x-parent and x-depth parameters. I think that will solve your issue.

**Reply by zhucan** [reply]
Is there example for it？

**Reply by louwrentius** [reply]
> Is there example for it？

In the README section "Comparing two or more benchmarks in a single line chart"

**Reply by zhucan** [reply]
if we want to compare all device, we need to specify the output dir  one by one? it's very inconvenient.

**Reply by louwrentius** [reply]
Yes for each device you have to specify the input directory. 

You can also use an INI file for fio-plot if that helps.

If you have any suggestions on how to improve this (how it should work) let me know.
